tests/jballerina-unit-test/src/test/resources/test-src/types/integer/integer-value.bal

```diff
@@ -109,3 +109,12 @@ function testIntSubtypeField() {
     test:assertEquals(val?.minutes, 9);
     test:assertEquals(val?.seconds, <decimal>8.555);
 }
+
+function testStaticTypeOfUnaryExpr() {
```

Comment:
Dont we need to check the static type of the result of unary expr here? We are checking the result of the type-cast expr right?

Reply:
we are checking the static type with the error tests right... these tests were actually additional just to see if we can make the scenario work with casting.. will change the function name

Reply:
Since these don't work in update 2 as well I have removed these tests. These should work once the singleton typing is properly implemented.